app/utils/fileDownload/index.ts

```diff
@@ -0,0 +1,59 @@
+import RNFetchBlob, { FetchBlobResponse } from 'rn-fetch-blob';
+import FileViewer from 'react-native-file-viewer';
+
+import EventEmitter from '../events';
+import { LISTENER } from '../../containers/Toast';
+import I18n from '../../i18n';
+import { DOCUMENTS_PATH, DOWNLOAD_PATH } from '../../constants/localPath';
+
+interface IAttachment {
+	title: string;
+	title_link: string;
+	type: string;
+	description: string;
+}
+
+export const getExtensionType = (text: string): string | undefined => text.split('.').pop();
+
+export const getLocalFilePathFromFile = (localPath: string, attachment: IAttachment): string => {
+	const fileName = attachment.title.split('.')[0];
+	return `${localPath}${fileName}.${getExtensionType(attachment.title_link)}`;
+};
+
+export const fileDownload = (url: string, attachment: IAttachment): Promise<FetchBlobResponse> => {
+	const path = getLocalFilePathFromFile(DOWNLOAD_PATH, attachment);
+
+	const options = {
+		path,
+		timeout: 10000,
+		indicator: true,
+		overwrite: true,
+		addAndroidDownloads: {
+			path,
+			notification: true,
+			useDownloadManager: true
+		}
+	};
+
+	return RNFetchBlob.config(options).fetch('GET', url);
+};
+
+export const fileDownloadAndPreview = async (url: string, attachment: IAttachment): Promise<void> => {
+	const path = getLocalFilePathFromFile(DOCUMENTS_PATH, attachment);
+	const file = await RNFetchBlob.config({
+		timeout: 10000,
+		indicator: true,
+		path
+	}).fetch('GET', url);
+
+	if (file) {
```

Comment:
What if we don't have a file?
Shouldn't we try catch `RNFetchBlob.config` and return a proper error?

Reply:
Done!